dbt Core CLI hints (#9612)
fishtown-analytics/docs.getdbt.com: website/docs/reference/global-configs/hints.md

---
title: "CLI hints"
id: "hints"
sidebar: "CLI hints"
---

<VersionCallout version="1.12" />

dbt surfaces occasional, non-blocking hints during command execution that point toward ways to reduce build and parse times. Hints appear as `[HINT]`-prefixed notes in the CLI output; they are informational only and never block a run.

> Each hint is shown at most once per week per project target directory.

Reviewer: ```suggestion
Each hint is shown, at most, once per week per project target directory.
```

## Disabling hints

Hints are enabled by default. To disable them, use any of the following:

- **CLI**: `--no-hints-enabled`
- **Environment variable**: `DBT_ENGINE_HINTS_ENABLED=false`
- **Project file**: Set `hints_enabled: false` in `dbt_project.yml`:
    <File name='dbt_project.yml'>

    ```yaml
    flags:
      hints_enabled: false
    ```

    </File>

> ## Available hints

Reviewer: Can we put a lead in sentence here to avoid back-to-back headers? Something like:
```suggestion
## Available hints

The following hints are available today, with plans to add more in the future. 
```

---
title: "CLI hints"
id: "hints"
sidebar: "CLI hints"
---

<VersionCallout version="1.12" />

dbt surfaces occasional, non-blocking hints during command execution that point toward ways to reduce build and parse times. Hints appear as `[HINT]`-prefixed notes in the CLI output; they are informational only and never block a run.

Each hint is shown, at most, once per week per project target directory.

## Disabling hints

Hints are enabled by default. To disable them, use any of the following:

- **CLI**: `--no-hints-enabled`
- **Environment variable**: `DBT_ENGINE_HINTS_ENABLED=false`
- **Project file**: Set `hints_enabled: false` in `dbt_project.yml`:
    <File name='dbt_project.yml'>

    ```yaml
    flags:
      hints_enabled: false
    ```

    </File>

## Available hints

> The following hints are available today, with plans to add more in the future. 

Author: ```suggestion
The following hints are available starting <Constant name="core" />  v1.12.
```
